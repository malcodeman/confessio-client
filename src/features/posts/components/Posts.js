import React, { Component } from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import Post from "./Post";
import { getPosts } from "../actions/posts_actions";

const PostsSection = styled.section`
  padding: 20px;
  max-width: 992px;
  margin: 0 auto;
`;

class Posts extends Component {
  componentDidMount = () => {
    this.props.getPosts();
  };
  render() {
    return (
      <PostsSection>
        {this.props.posts.length > 0 ? (
          this.props.posts.map(post => {
            return (
              <Post key={post.id} text={post.text} createdAt={post.createdAt} />
            );
          })
        ) : (
          <p>No posts.</p>
        )}
      </PostsSection>
    );
  }
}

const mapStateToProps = state => {
  return {
    posts: state.posts.posts
  };
};

const mapDispatchToProps = dispatch => {
  return {
    getPosts: () => dispatch(getPosts())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Posts);
