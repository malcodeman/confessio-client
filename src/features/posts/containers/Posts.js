import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { Link } from "react-router-dom";

import PostLoading from "../components/PostLoading";
import Post from "../components/Post";
import NewPostForm from "./PostsNewForm";
import footerLinks from "../data/footerLinks";

import {
  getPosts,
  createComment,
  createLike,
  destroyLike,
  createBookmark,
  destroyBookmark
} from "../actions/postsActionCreators";
import Suggestions from "../components/Suggestions";

const Container = styled.div`
  display: grid;
  padding: 0 32px;
  @media (min-width: 768px) {
    padding: 0;
  }
  @media (min-width: 992px) {
    grid-template-columns: 2fr 1fr;
    grid-gap: 24px;
  }
`;

const PostsContainer = styled.div`
  margin: 0 auto;
  width: 100%;
  max-width: 576px;
  @media (min-width: 992px) {
    max-width: initial;
  }
`;

const SidebarContainer = styled.div`
  display: none;
  @media (min-width: 992px) {
    display: flex;
    flex-direction: column;
  }
`;

const Sidebar = styled.div`
  position: sticky;
  top: 88px;
`;

const Footer = styled.footer``;

const List = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 24px;
`;

const ListItem = styled.li`
  :not(:last-child) {
    margin-right: 4px;
  }
`;

const StyledLink = styled(Link)`
  font-size: 0.8rem;
  color: ${props => props.theme.secondary};
`;

const Copyright = styled.span`
  font-size: 0.8rem;
  text-transform: uppercase;
  color: ${props => props.theme.secondary};
`;

class Posts extends React.Component {
  componentDidMount = () => {
    const { getPosts } = this.props;

    getPosts();
  };

  render() {
    const {
      posts,
      me,
      loading,
      createComment,
      createBookmark,
      destroyBookmark,
      createLike,
      destroyLike
    } = this.props;

    return (
      <Container>
        <PostsContainer>
          <NewPostForm />
          {posts.length === 0 && loading && <PostLoading />}
          {posts.map(post => {
            return (
              <Post
                key={post.id}
                post={post}
                me={me}
                createComment={createComment}
                createBookmark={createBookmark}
                createLike={createLike}
                destroyLike={destroyLike}
                destroyBookmark={destroyBookmark}
              />
            );
          })}
        </PostsContainer>
        <SidebarContainer>
          <Sidebar>
            <Suggestions />
            <Footer>
              <List>
                {footerLinks.map(link => {
                  return (
                    <ListItem key={link.label}>
                      <StyledLink to={link.route}>{link.label}</StyledLink>
                    </ListItem>
                  );
                })}
              </List>
              <Copyright>Â© Polaroid</Copyright>
            </Footer>
          </Sidebar>
        </SidebarContainer>
      </Container>
    );
  }
}

const mapStateToProps = state => {
  return {
    posts: state.posts.posts,
    loading: state.posts.loading,
    me: state.users.me
  };
};

export default connect(
  mapStateToProps,
  {
    getPosts,
    createComment,
    createLike,
    destroyLike,
    createBookmark,
    destroyBookmark
  }
)(Posts);
