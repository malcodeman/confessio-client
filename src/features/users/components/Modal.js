import React from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Modal = props => {
  return <Wrapper>{props.children}</Wrapper>;
};

export default Modal;
