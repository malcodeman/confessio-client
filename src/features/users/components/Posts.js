import React from "react";
import styled from "styled-components";

const Grid = styled.div`
  display: grid;
  grid-gap: 4px;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: auto;
  @media (min-width: 768px) {
    grid-gap: 10px;
  }
`;

const Post = styled.img`
  max-width: 100%;
  height: 100%;
  object-fit: cover;
`;

const Posts = props => {
  return (
    <Grid>
      {props.posts.map(post => {
        return <Post key={post.id} src={post.photoURL} />;
      })}
    </Grid>
  );
};

export default Posts;
