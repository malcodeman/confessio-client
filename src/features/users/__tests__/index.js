import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Modal from "../components/Modal";
import Posts from "../components/Posts";

test("Modal matches snapshot", () => {
  const component = renderer.create(<Modal />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Posts(no posts) matches snapshot", () => {
  const component = renderer.create(<Posts />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const posts = [
  {
    id: 1,
    photoURL:
      "https://instagram.fsjj1-1.fna.fbcdn.net/vp/d290920f9a938d426f6edd3bfdf3a4e0/5BAB4F61/t51.2885-15/e35/33630509_2122355564708396_6000352995101900800_n.jpg"
  },
  {
    id: 2,
    photoURL:
      "https://instagram.fsjj1-1.fna.fbcdn.net/vp/61faee47738a31b77346f3d524354d18/5BB89C6E/t51.2885-15/e35/33177999_204623263489562_6594454715688288256_n.jpg"
  }
];

test("Posts matches snapshot", () => {
  const component = renderer.create(<Posts posts={posts} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
