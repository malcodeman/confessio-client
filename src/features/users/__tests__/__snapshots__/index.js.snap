// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<div
  className="sc-bdVaJa eLBSzQ"
/>
`;

exports[`Posts matches snapshot 1`] = `
<div
  className="sc-bwzfXH iZrXVz"
>
  <img
    className="sc-htpNat brnEhT"
    src="https://instagram.fsjj1-1.fna.fbcdn.net/vp/d290920f9a938d426f6edd3bfdf3a4e0/5BAB4F61/t51.2885-15/e35/33630509_2122355564708396_6000352995101900800_n.jpg"
  />
  <img
    className="sc-htpNat brnEhT"
    src="https://instagram.fsjj1-1.fna.fbcdn.net/vp/61faee47738a31b77346f3d524354d18/5BB89C6E/t51.2885-15/e35/33177999_204623263489562_6594454715688288256_n.jpg"
  />
</div>
`;

exports[`Posts(no posts) matches snapshot 1`] = `
<div
  className="sc-bxivhb efivZG"
>
  No posts yet
</div>
`;
